Sources/Lumina/UI/LuminaButton.swift

```diff
@@ -0,0 +1,189 @@
+  private var squareSystemButtonHeight = 40
+  private var cancelButtonWidth = 70
+  private var cancelButtonHeight = 30
+  private var shutterButtonDimension = 70
+  private var style: SystemButtonType?
+  private var symbolConfig = UIImage.SymbolConfiguration(pointSize: 25)
+  private var border: UIView?
+  private var _image: UIImage?
+  var image: UIImage? {
+    get {
+      return _image
+    }
+    set {
+      self.setImage(newValue, for: UIControl.State.normal)
+      _image = newValue
+    }
+  }
+
+  private var _text: String?
+  var text: String? {
+    get {
+      return _text
+    }
+    set {
+      self.setTitle(newValue, for: UIControl.State.normal)
+      _text = newValue
+    }
+  }
+
+  required init() {
+    super.init(frame: CGRect.zero)
+    self.backgroundColor = UIColor.clear
+    if let titleLabel = self.titleLabel {
+      titleLabel.textColor = UIColor.white
+      titleLabel.font = UIFont.systemFont(ofSize: 20)
+      titleLabel.textAlignment = .center
+    }
+  }
+
+  init(with systemStyle: SystemButtonType) {
+    super.init(frame: CGRect.zero)
+    self.style = systemStyle
+    self.backgroundColor = UIColor.clear
+    if let titleLabel = self.titleLabel {
+      titleLabel.textColor = UIColor.white
+      titleLabel.font = UIFont.systemFont(ofSize: 20)
+    }
+    switch systemStyle {
+      case .torch:
+        self.image = UIImage(systemName: "bolt.slash.fill", withConfiguration: symbolConfig)?.withTintColor(.white, renderingMode: .alwaysOriginal)
+        self.frame = CGRect(origin: CGPoint(x: 10, y: 10), size: CGSize(width: self.squareSystemButtonWidth, height: self.squareSystemButtonHeight))
+      case .cameraSwitch:
+        self.image = UIImage(systemName: "arrow.triangle.2.circlepath.camera.fill", withConfiguration: symbolConfig)?.withTintColor(.white, renderingMode: .alwaysOriginal)
+        self.frame = CGRect(origin: CGPoint(x: UIScreen.main.bounds.maxX - 50, y: 10), size: CGSize(width: self.squareSystemButtonWidth, height: self.squareSystemButtonHeight))
+      case .cancel:
+        self.image = UIImage(systemName: "xmark.circle.fill", withConfiguration: symbolConfig)?.withTintColor(.white, renderingMode: .alwaysOriginal)
+        self.frame = CGRect(origin: CGPoint(x: UIScreen.main.bounds.maxX - 50, y: 10), size: CGSize(width: self.squareSystemButtonWidth, height: self.squareSystemButtonHeight))
+      case .shutter:
+        self.backgroundColor = UIColor.normalState
+        var minY = UIScreen.main.bounds.maxY
+        if #available(iOS 11, *) {
+          minY = self.safeAreaLayoutGuide.layoutFrame.maxY
+        }
+        minY -=  80
+        self.frame = CGRect(origin: CGPoint(x: UIScreen.main.bounds.midX - 35, y: minY), size: CGSize(width: self.shutterButtonDimension, height: self.shutterButtonDimension))
+        self.layer.cornerRadius = CGFloat(self.shutterButtonDimension / 2)
+        self.layer.borderWidth = 3
+        self.layer.borderColor = UIColor.borderNormalState
+      default:
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dokun1/Lumina/blob/f3c4aa8d5c66dd9d2660cdba8e658b8902c9752f/Sources/Lumina/UI/LuminaButton.swift#L96"></span>Case statements should vertically align with their enclosing switch statement.<br />
<code>switch_case_alignment</code> <code>LuminaButton.swift:96</code></td>
    </tr>
  </tbody>
</table>